classes/observer.php

```diff
@@ -135,4 +137,334 @@ public static function mod_quiz_event_attempt_submitted(
+        ];
+
+        $cl->create_group($data, $courseId);
+    }
+
+    /**
+     * Group deleted event handler.
+     * @param \core\event\group_deleted $event
+     */
+    public static function group_deleted(\core\event\group_deleted $event) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+        
+        $eventdata = $event->get_data();
+        $cl = new \plagiarism_copyleaks_comms();
+        $courseId = $eventdata['courseid'];
+        $data = (array)[
+            'groupId' => $eventdata['objectid'],
+        ];
+        $cl->delete_group($data, $courseId);
+    }
+
+
+    /**
+     * Group updated event handler.
+     * @param \core\event\group_updated $event
+     */
+    public static function group_updated(\core\event\group_updated $event) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+
+        $eventdata = $event->get_data();
+        $cl = new \plagiarism_copyleaks_comms();
+        $groupname = groups_get_group_name($eventdata['objectid']);
+        $courseId = $eventdata['courseid'];
+        $data = (array)[
+            'groupId' => $eventdata['objectid'],
+            'groupName' => $groupname,
+        ];
+
+        $cl->update_group($data, $courseId);
+    }
+
+    /**
+     * Group member added event handler.
+     * @param \core\event\group_member_added $event
+     */
+    public static function group_member_added(\core\event\group_member_added $event) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+            
+        $eventdata = $event->get_data();
+        $cl = new \plagiarism_copyleaks_comms();
+        $user = get_complete_user_data('id', $eventdata['relateduserid']);
+        $groupname = groups_get_group_name($eventdata['objectid']);
+        $courseId = $eventdata['courseid'];
+        $groupdata = (array)[
+            'groupId' => $eventdata['objectid'],
+            'groupName' => $groupname,
+        ];
+
+        $userdata = (array)[
+            'mppUserId' => $eventdata['relateduserid'],
+            'userName' => $user->firstname . " " . $user->lastname,
+            'userEmail' => $user->email,
+        ];
+
+        $data = (array)[
+            'group' => $groupdata,
+            'user' => $userdata
+        ];
+
+        $cl->add_group_member($data, $courseId);
+    }
+
+    /**
+     * Group member removed event handler.
+     * @param \core\event\group_member_removed $event
+     */
+    public static function group_member_removed(\core\event\group_member_removed $event) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+
```

Comment:
@ShadeAma shouldn't we check if assign is enabled here? same as other events

if (!plagiarism_copyleaks_moduleconfig::is_module_enabled('assign', $cmid)) {
  return;
}